### PR TITLE
Update to documentation of Featurizers.

### PR DESCRIPTION
**Pull request by ARY2260**
# Pull Request Template

## Description

A brief introduction to Featurizer abstract class, specific featurizer classes and data classes.

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rbharath** [comment]
@ARY2260 Sorry for the late review! It looks like the docs build is failing. Can you come by the dev call or office hours this week for us to review and figure out what's happening?

**Comment by ARY2260** [comment]
> @ARY2260 Sorry for the late review! It looks like the docs build is failing. Can you come by the dev call or office hours this week for us to review and figure out what's happening?

Yup sure. I will be there during this week's dev call.

**Comment by ARY2260** [comment]
I have added jinja2<3.1 in requirements.txt. Will it be possible to run the tests again? The requirements.txt was executed in the step prior to the step where the error occurred. So, this change would install an older version of jinja2, hence should solve the error.

**Comment by ARY2260** [comment]
#2892

**Comment by arunppsg** [comment]
Thanks for the fix. On a side note, the error in doc building is not related to this PR! DeepChem do not use `jinja` template directly, but `sphinx` uses `jinja` to build docs. A new version of `jinja` (3.0) was released which removed depreciated code causing the failure in sphinx.

File: docs/source/api_reference/featurizers.rst [hunk 1]
```diff
@@ -31,6 +31,10 @@ Molecule Featurizers
 
 These featurizers work with datasets of molecules.
 
+All featurizers are based on an abstract parent class (:code:`dc.feat.Featurizer`). This is used to create abstract base classes for a few general types of featurization, including (:code:`MolecularFeaturizer`), (:code:`MaterialCompositionFeaturizer`), etc.
```
Comment (arunppsg) [inline docs/source/api_reference/featurizers.rst:34]: I think these will be more apt in the `Base Featurizers (for develop)` section of docs, just above `MolecularFeaturizer` section (https://github.com/deepchem/deepchem/blob/master/docs/source/api_reference/featurizers.rst#molecularfeaturizer)

**Comment by rbharath** [comment]
Thanks for the fix @ARY2260! Once @arunppsg's comment is addressed we should be good to merge

**Comment by ARY2260** [comment]
> Thanks for the fix. On a side note, the error in doc building is not related to this PR! DeepChem do not use `jinja` template directly, but `sphinx` uses `jinja` to build docs. A new version of `jinja` (3.0) was released which removed depreciated code causing the failure in sphinx.

So there is no need for the fix now?

**Comment by rbharath** [comment]
No the fix is still needed :). @arunppsg was just mentioning that the error you fixed wasn't caused by your code change.

**Pull request closed without merging**